src/client/sw/middleware/fetch.js

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { expiration, cacheRouter } from '@americanexpress/one-service-worker';
+
+import {
+  oneAppRegExp, moduleRegExp, langPackRegExp,
+} from './utility';
+
+export default function createFetchMiddleware() {
+  return [
+    {
+      cacheName: 'one-app-cache',
+      // eslint-disable-next-line no-useless-escape
+      match: oneAppRegExp,
+    },
+    {
+      cacheName: 'module-cache',
```

Comment:
I have to ask for sanity sake: How will the cache know that a new module has been added and clear the stale module?

Reply:
There is an invalidation middleware planned, that will support removing modules when their version is changed.